blazar/manager/service.py

```diff
@@ -749,7 +749,7 @@ def _create_reservation(self, values):
         db_api.reservation_update(reservation['id'],
                                   {'resource_id': resource_id})
 
-    def _allocation_candidates(self, lease, reservations):
+    def _allocation_candidates(self, lease, reservations, can_retry=True):
```

Comment:
The `@retrying.retry` decorator might also work here, if you set the max attempts to 2. It would be a bit clunky... reading through this function maybe it's not possible as written.